lib/router/src/lib.rs

```diff
@@ -39,22 +39,39 @@ pub trait Recognize<Request> {
     fn recognize(&self, req: &Request) -> Option<Self::Target>;
 }
 
+pub trait Make<Target> {
```

Comment:
might be overly nitpicky but i'm not a huge fan of having a trait that's just called `Make`, esp. in a world where we already have a bunch of types and traits named `MakeFoo`...